### ISSUE TITLE
Drag and Drop an External File in QML

### ISSUE THREAD

**Issue by jaruba** [RSATom/WebChimera#88]
I would need to be able to drag and drop an external file in qml and get the path. I tried ```DropArea``` for this and it didn't work (I think it works only for internal qml elements).

**Comment by RSATom** [comment]
checked, it works fine in standalone qml application, but by some reason does not works with plugin.

**Comment by RSATom** [comment]
It's Qt limitation. There are no way to enable drag and drop for not top level windows. Patch to Qt needed.

**Comment by jaruba** [comment]
"drag and drop for not top level windows"

Can you explain this a bit? Do you think that if I focus the node-webkit window when I'm hovering with a dragging file it might work to drop it and get the file path?

**Comment by RSATom** [comment]
No, it will not help. I mean drag and drop works only if top level (or by another words main application window) was created by Qt. But in our case top level window is browser window.

**Comment by jaruba** [comment]
So it doesn't work because the plugin is a child process of node-webkit? Then I wonder if this is not a OS limitation.

If so, I wonder if there would be any possible way to hack it as I have the possibility of a drop event in html.. I'm wondering what will happen if I attach the drop event to the plugin object tag or maybe encapsulate the entire page into an iframe and set the drop event on the iframe.

**Comment by RSATom** [comment]
> So it doesn't work because the plugin is a child process of node-webkit? Then I wonder if this is not a OS limitation.

It doesn't work because plugin window is child window on window created not by Qt. And yes, it's not OS limitation, it's Qt limitation and could be fixed. I just don't know how hard it could be since problem is in Qt Core.

> If so, I wonder if there would be any possible way to hack it as I have the possibility of a drop event in html.. I'm wondering what will happen if I attach the drop event to the plugin object tag or maybe encapsulate the entire page into an iframe and set the drop event on the iframe.

Don't know. It's better just try.

**Comment by jaruba** [comment]
Tried adding the drop event to the plugin object then the entire window, works on everything except the plugin area. :(

**Comment by RSATom** [comment]
I'll try to solve this problem, but only after next release and #89

**Comment by jaruba** [comment]
Also, I think [this patch](https://codereview.qt-project.org/#/c/88612/1/src/quick/items/qquickdroparea.cpp) mentioned on [this stackoverflow answer](http://stackoverflow.com/questions/22787242/reject-external-files-dragged-in-a-droparea-without-breaking-the-droparea#answer-28800328) only 16 days ago should be added in the change too, as this is a recent bug that has not yet been patched and it is related to an issue with rejecting files in DropArea. Won't fix the dropping of external files, but it will fix a future issue that might arise.

**Comment by RSATom** [comment]
It's very strange this bug not fixed so long time.

**Comment by jaruba** [comment]
Indeed, but the patch seems to be authentic as it's mentioned on the answer from stackoverflow and the official bug's page..

**Comment by jaruba** [comment]
This was only for NW.js and https://github.com/RSATom/WebChimera.js is the clean solution for this.
